db/db_sst_test.cc

```diff
@@ -1646,34 +1649,24 @@ TEST_F(DBSSTTest, GetTotalSstAndBlobFilesSize) {
     }
     ASSERT_OK(Flush());
   }
-  ASSERT_EQ("5", FilesPerLevel(0));
-  ASSERT_EQ(50, GetBlobFileNumbers().size());
+  Close();
 
-  std::vector<LiveFileMetaData> live_files_meta;
-  dbfull()->GetLiveFilesMetaData(&live_files_meta);
-  ASSERT_EQ(live_files_meta.size(), 5);
+  bool is_get_file_size_called = false;
+  SyncPoint::GetInstance()->SetCallBack(
+      "MockFileSystem::GetFileSize:CheckFileType", [&](void* arg) {
+        std::string* filename = reinterpret_cast<std::string*>(arg);
+        if (filename->find(".blob") != std::string::npos) {
```

Comment:
I didn't check `.sst` here. GetFileSize() still be invoked for these files (`.sst`) if the user didn't set `options.skip_checking_sst_file_sizes_on_db_open`.

```
12         0x1081d89f4 rocksdb::CompositeEnv::GetFileSize(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned long long*) + 112
13         0x107d4d8a4 rocksdb::DBImpl::CheckConsistency() + 1712
14         0x107e6e4a4 rocksdb::DBImpl::Recover(std::__1::vector<rocksdb::ColumnFamilyDescriptor, std::__1::allocator<rocksdb::ColumnFamilyDescriptor> > const&, bool, bool, bool, unsigned long long*) + 3120
15         0x107e755ac rocksdb::DBImpl::Open(rocksdb::DBOptions const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::vector<rocksdb::ColumnFamilyDescriptor, std::__1::allocator<rocksdb::ColumnFamilyDescriptor> > const&, std::__1::vector<rocksdb::ColumnFamilyHandle*, std::__1::allocator<rocksdb::ColumnFamilyHandle*> >*, rocksdb::DB**, bool, bool) + 1452
16         0x107e74f50 rocksdb::DB::Open(rocksdb::DBOptions const&, std::__1::basic_string<char, std::
```